nginx.tmpl

```diff
@@ -97,16 +102,17 @@ upstream {{ $host }} {
 {{ $certName := (first (groupByKeys $containers "Env.CERT_NAME")) }}
 
 {{/* Get the best matching cert  by name for the vhost. */}}
-{{ $vhostCert := (closest (dir "/etc/nginx/certs") (printf "%s.crt" $host))}}
+{{ $vhostCert := (closest (dir "/etc/nginx/certs") (printf "%s.key" $host))}}
```

Comment:
This entire vhost matching section is a bit off, tbh. I'll write up a separate issue/PR once I get the chance.

The three `{{ $vhostCert := replace $vhostCert ".[....] " "" -1 }}` lines below this look like they would break if the vhost name ends in .pem, .crt, or .key (for whatever reason). I.e. if the vhost name is `foo.key`, and the certificate file name is `foo.key.crt`, then the name could get stripped down to `foo`. I'll need to do a bit of testing to see if this is actually an issue, but I think the overall section needs a rework to cover these edge cases.